apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts

```diff
@@ -491,29 +516,26 @@ function processControlValuesBySchema(
 ): StepIssuesDto {
   let issues: StepIssuesDto = {};
 
-  const cleanedControlValues = controlValues ? cleanObject(controlValues) : {};
-
-  if (!controlSchema || !cleanedControlValues) {
+  if (!controlSchema || !controlValues) {
     return issues;
   }
 
   const ajv = new Ajv({ allErrors: true });
   addFormats(ajv);
   const validate = ajv.compile(controlSchema);
-  const isValid = validate(cleanedControlValues);
+  const isValid = validate(controlValues);
```

Comment:
This validation logic is duplicated between upsert and preview. Should we consider consolidating it into a validate workflow use case?

In Upsert, we want this logic to generate 422 errors or issues. In Preview, we need this logic to fill in the invalid controls with defaults. 

It feels like both functionalities can be handled by a reusable use case that only handles the AJV part and returns errors that are handled accordingly per use case.

Reply:
agree :)
i will create task for it.